pkg/apis/kops/validation/legacy.go

```diff
@@ -76,14 +76,24 @@ func ValidateCluster(c *kops.Cluster, strict bool) *field.Error {
 
 	requiresSubnets := true
 	requiresNetworkCIDR := true
+	requiresSubnetCIDR := true
 	switch kops.CloudProviderID(c.Spec.CloudProvider) {
 	case kops.CloudProviderBareMetal:
 		requiresSubnets = false
 		requiresNetworkCIDR = false
+		if c.Spec.NetworkCIDR != "" {
+			return field.Invalid(fieldSpec.Child("NetworkCIDR"), c.Spec.NetworkCIDR, "NetworkCIDR should not be set on bare metal")
```

Comment:
As the NetworkCIDR is ignored, makes more sense to error if it is set

Reply:
Can we move this to another PR?  I love it, but it kinda does not apply to the work you are doing.